File: 1005-Maximize Sum Of Array After K Negations-K 次取反后最大化的数组和/1005-Maximize Sum Of Array After K Negations-K 次取反后最大化的数组和.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
 var largestSumAfterKNegations = function(nums, k) {
    let res = 0
    nums.sort((a,b)=>a-b)
    for(let i = 0; i < nums.length && k > 0; i++){
        if(nums[i] <= 0){
            nums[i] = -nums[i]
            k--
        }else{
            k = k % 2 === 1 ? 1 : 0
            break
        }
    }
    nums.sort((a,b)=>a-b)
    if(k === 1)nums[0] = -nums[0]
    for(let i = 0; i < nums.length; i++){
        res += nums[i]
    }
    return res
};

File: 1005-Maximize Sum Of Array After K Negations-K 次取反后最大化的数组和/1005-Maximize Sum Of Array After K Negations-K 次取反后最大化的数组和.java
class Solution {
    public int largestSumAfterKNegations(int[] nums, int K) {
    int res = 0;
    Arrays.sort(nums);
    for(int i = 0; i < nums.length && K > 0; i++){
        if(nums[i] <= 0){
            nums[i] = -nums[i];
            K--;
        }else{
            K = K % 2 == 1 ? 1 : 0;
            break;
        }
    }
    Arrays.sort(nums);
    if(K == 1)nums[0] = -nums[0];
    for(int i = 0; i < nums.length; i++){
        res += nums[i];
    }
    return res;
    }
}